_knitr/A Shiny app for Hong Kong District Councillors_20200801.md

```diff
@@ -1,50 +1,204 @@
+
+Once we settled on how the map looked in relation to the Shiny app, we then spent some additional time and effort to investigate using [leaflet](https://github.com/rstudio/leaflet). The reason for moving to **leaflet** maps because of their interactivity: we understood our users would want to explore the HK map interactively to find out what consituency they belong to or to find out one that was of interest. This was because we were aware that people may know what region they live in but they may not know the name of the consituency.
+
+## 💭 What are our next steps?
+
+There were some cool features that we would have liked to, but have not been able to implement:  
+
+- [precommit hooks](https://github.com/Hong-Kong-Districts-Info/dashboard-hkdistrictcouncillors/issues/17): Those familiar with Python may be aware of pre-commit hooks as ways to automatically detect whether your repo contains anything sensitive like a `.secrets` file. Setting this up will enable us to have automated checks run each time we make a commit to assure we are follow specified standards.
+    + Unfortunately, we named our repo with a hyphen so the pre-commit hooks won't work.
+    + [codecov](https://github.com/Hong-Kong-Districts-Info/dashboard-hkdistrictcouncillors/issues/33): Allows us to robustly test the functions in our code so that they work under a multitude of scenarios such as when users encounter problems.
+- [modularise shiny code](https://github.com/Hong-Kong-Districts-Info/dashboard-hkdistrictcouncillors/issues/26): Ensures our Shiny code is more easier to follow.
+- [language selection](https://github.com/Hong-Kong-Districts-Info/dashboard-hkdistrictcouncillors/issues/36): Currently the app is a smorgasbord of English and Chinese. Consequently, it looks messy. We want to implement the ability for the user to choose which language they want to see the app in and the app's language will update accordingly.
+- Release to alpha testers to get early feedback.
+- More or our enhancements / spikes are listed here on [GitHub](https://github.com/Hong-Kong-Districts-Info/dashboard-hkdistrictcouncillors/issues)
+
+One of the things that we wanted to try out with this open-source project is to adhere to some DevOps best practices, yet unfortunately some of these were either easier to set up from the beginning, or require more time and knowledge (on our part) to set up. As we develop a V2 of this Shiny App and work on [other projects](https://hong-kong-districts-info.github.io/portfolio/), we hope to find the opportunity to implement more of the above features.
+
+## 🔥 Other features in the app
+
+There were also a number of features that we have implemented, but were not detailed in this post. For instance:
+
+- Adding a searchable DataTable with information on the District Councillors, with the **DT** package
+- Embedding a user survey within the Shiny app
+- Adding a tutorial to go through features of the Shiny app, using the **shinyhelper** package
```

Comment:
Do you mean the individual help icons which is **shinyhelper**? Or the interactive tutorial which is **rintrojs**?

Reply:
Oh yes - good spot